invenio_users_resources/notifications.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-Users-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""User specific resources for notifications."""
+
+
+from invenio_records_resources.notifications import RecipientFilter
+
+
+class UserPreferencesRecipientFilter(RecipientFilter):
+    """Recipient filter for notifications being enabled at all."""
+
+    def run(self, recipients, **kwargs):
```

Comment:
It might make sense to add a unit test for this: this might help to prevent sending unwanted notifications around, in case of code mistakes.

Reply:
added